file.go

```diff
@@ -86,8 +86,37 @@ func extractFiles(zipFilePath, filesToExtractFromZipPath, localPath string) erro
 	// printing some of their contents.
 	for _, f := range r.File {
 
-		// If the given file is in the filesToExtractFromZipPath, proceed
-		if strings.Index(f.Name, pathPrefix) == 0 {
+			//
+			// Skip the current archive item being processed based on
+			// rules described in comments below.
+			//
+			if f.FileInfo().IsDir() {
+				// The current archive item is a directory.
+				// Archive item's f.Name will always be appended with a "/", so we use
+				// that fact to ensure we are working with a full directory name. Skip
+				// to next item if (pathPrefix + "/") is not a prefix in f.Name
+				if strings.Index(f.Name, pathPrefix + "/") != 0 {
+					continue
+				}
+			} else {
+				// The current archive item is a file.
+				// There are three things possible here:
+				//		1  User specified a filename using --source-path option and if we hit
+				//		   that file in archive, we need to process this file, so do not skip.
```

Comment:
In the PR, the indentation here seems off?

Reply:
@brikis98 , great suggestions. Agree that `!=`'s are harder to mentally reason about. Also agree that extracting the code into the check into a method would be helpful, it would convey the intention of the code much more clearly.

The method name `shouldExtractPathInZip` sounds good and the compact code you suggested as its contents look great.

Once I incorporate the call to the new method into `extractFiles` method, the indentation should fall into place.

I will work on this and push an updated PR soon.

Reply:
I have extracted the check condition to `shouldExtractPathInZip` method.

There is a slight modification in the one-line return statement you suggested. Unfortunately, there is no `IsFile()` method available on `FileInfo()`, so I set a variable `zipPathIsFile := !zipPath.FileInfo().IsDir()` and then use it in `return` statement.

PTAL